Dockerfile

```diff
@@ -1,16 +1,18 @@
 # syntax=docker/dockerfile:1
 
 ARG BASE_VARIANT=alpine
+ARG BASE_DEBIAN_DISTRO=bookworm
+ARG BASE_ALPINE_VERSION=3.17
```

Comment:
Curious; was there a reason for renaming the arg? If there's no specific reason, I'd like to keep it the same (also consistent with the moby repo where we use `ALPINE_VERSION`

Also considering if we should use `_VERSION` for debian as well, but I see we use `BASE_DEBIAN_DISTRO` in moby. 

Also can you put the alpine version above the debian version? I think we may update the alpine version more often, and having it "separate" from the GO_VERSION can potentially reduce conflicts when cherry-picking go version updates to release branches.